File: round1_full_pack_optionA_solutions/E1_Palindrome_Checker/Solutions/Solution.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    int T;
    cin >> T;

    while (T--) {
        string s;
        cin >> s;

        int n = s.length();
        bool ok = true;

        for (int i = 0; i < n / 2; i++) {
            if (s[i] != s[n - i - 1]) {
                ok = false;
                break;
            }
        }

        if (ok)
            cout << "Yes\n";
        else
            cout << "No\n";
    }

    return 0;
}

File: round1_full_pack_optionA_solutions/E1_Palindrome_Checker/Solutions/Solution.py
import sys

T = int(sys.stdin.readline())

for _ in range(T):
    s = sys.stdin.readline().strip()
    n = len(s)
    ok = True

    for i in range(n // 2):
        if s[i] != s[n - i - 1]:
            ok = False
            break

    if ok:
        print("Yes")
    else:
        print("No")
